test/integration.spec.js

```diff
@@ -0,0 +1,198 @@
+          {is_array: false, is_dict: true, name: "plane_type", type: "STR"},
+          {is_array: false, is_dict: true, name: "plane_manufacturer", type: "STR"},
+          {is_array: false, is_dict: false, name: "plane_issue_date", type: "DATE"},
+          {is_array: false, is_dict: true, name: "plane_model", type: "STR"},
+          {is_array: false, is_dict: true, name: "plane_status", type: "STR"},
+          {is_array: false, is_dict: true, name: "plane_aircraft_type", type: "STR"},
+          {is_array: false, is_dict: true, name: "plane_engine_type", type: "STR"},
+          {is_array: false, is_dict: false, name: "plane_year", type: "SMALLINT"},
+          {is_array: false, is_dict: true, name: "origin_name", type: "STR"},
+          {is_array: false, is_dict: true, name: "origin_city", type: "STR"},
+          {is_array: false, is_dict: true, name: "origin_state", type: "STR"},
+          {is_array: false, is_dict: true, name: "origin_country", type: "STR"},
+          {is_array: false, is_dict: false, name: "origin_lat", type: "FLOAT"},
+          {is_array: false, is_dict: false, name: "origin_lon", type: "FLOAT"},
+          {is_array: false, is_dict: true, name: "dest_name", type: "STR"},
+          {is_array: false, is_dict: true, name: "dest_city", type: "STR"},
+          {is_array: false, is_dict: true, name: "dest_state", type: "STR"},
+          {is_array: false, is_dict: true, name: "dest_country", type: "STR"},
+          {is_array: false, is_dict: false, name: "dest_lat", type: "FLOAT"},
+          {is_array: false, is_dict: false, name: "dest_lon", type: "FLOAT"},
+          {is_array: false, is_dict: false, name: "origin_merc_x", type: "FLOAT"},
+          {is_array: false, is_dict: false, name: "origin_merc_y", type: "FLOAT"},
+          {is_array: false, is_dict: false, name: "dest_merc_x", type: "FLOAT"},
+          {is_array: false, is_dict: false, name: "dest_merc_y", type: "FLOAT"}
+        ])
+        done()
+      })
+    })
+  })
+
+  it(".query", done => {
+    const sql = "SELECT count(*) AS n FROM tweets_nov_feb WHERE country='CO'"
+    connector.connect((connectError, session) => {
+      expect(connectError).to.not.be.an("error")
+      session.query(sql, options, (error, data) => {
+        expect(connectError).not.be.an("error")
+        expect(Number(data[0].n)).to.equal(6400)
+        done()
+      })
+    })
+  })
+
+  it(".render", done => {
+    connector.connect((connectError, session) => {
+      expect(connectError).to.not.be.an("error")
+      session.renderVega(widgetId, vega, options, (renderVegaError, data) => {
+        expect(renderVegaError).to.not.be.an("error")
+        const imageData = convertToDataUrl(data.image)
+        expect(imageData, "should be a image data URL").to.match(imageRegex)
+        expect(imageData, "shouldn't be an empty image").to.not.match(emptyImageRegex)
+        done()
+      })
+    })
+  })
+
+  xit(".getResultRowForPixel", function (done) {
```

Comment:
commented out b/c I'm having trouble finding the right pixel coordinates ATM. Can merge without it IMO.